integration/spark/app/src/main/java/io/openlineage/spark/agent/EventEmitter.java

```diff
@@ -38,6 +41,9 @@ public EventEmitter(SparkOpenLineageConfig config, String applicationJobName)
     this.parentJobName = Optional.ofNullable(config.getParentJobName());
     this.parentJobNamespace = Optional.ofNullable(config.getParentJobNamespace());
     this.parentRunId = convertToUUID(config.getParentRunId());
+    this.rootParentJobName = Optional.ofNullable(config.getRootParentJobName());
```

Comment:
There are 3 parent* properties and 3 root* properties which are same. Shouldn't we introduce `JobAncestorIdentifier` to handle those?

Reply:
I don't understand why we need to complicate our internal api? If anything, I'd rather use lombok AllArgsConstructor/Builder to remove boilerplate.

Reply:
I don't think adding extra class to encapsulate three fields would bring complexity. 
But I am fine with this.

Reply:
I will refactor this in a future PR.